portable/NetworkInterface/TM4C/NetworkInterface.c

```diff
@@ -0,0 +1,780 @@
+                0);
+
+
+        // Clear any stray PHY interrupts that may be set.
+        ui16Val = MAP_EMACPHYRead(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_MISR1);
+        ui16Val = MAP_EMACPHYRead(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_MISR2);
+
+        // Configure and enable PHY interrupts
+        ui16Val = MAP_EMACPHYRead(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_SCR);
+        ui16Val |= (EPHY_SCR_INTEN_EXT | EPHY_SCR_INTOE_EXT);
+        MAP_EMACPHYWrite(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_SCR, ui16Val);
+
+        // Read the PHY interrupt status to clear any stray events.
+        ui16Val = MAP_EMACPHYRead(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_MISR1);
+
+        // Set MAC filtering options.  We receive all broadcast and mui32ticast
+        // packets along with those addressed specifically for us.
+        MAP_EMACFrameFilterSet(EMAC0_BASE, (EMAC_FRMFILTER_HASH_AND_PERFECT |
+                           EMAC_FRMFILTER_PASS_MULTICAST));
+
+        // Set the MAC address
+        MAP_EMACAddrSet(EMAC0_BASE, 0, &mac_address_bytes[0]);
+
+        // Clears any previously asserted interrupts
+        MAP_EMACIntClear(EMAC0_BASE, EMACIntStatus(EMAC0_BASE, false));
+
+        // Initialize the DMA descriptors
+        _dma_descriptors_init();
+
+        // Enable TX/RX
+        MAP_EMACTxEnable(EMAC0_BASE);
+        MAP_EMACRxEnable(EMAC0_BASE);
+
+        // Set the interrupt to a lower priority than the OS scheduler interrupts
+        MAP_IntPrioritySet(INT_EMAC0,  (6 << (8 - configPRIO_BITS)));
+
+        // Probe the PHY with the stack driver
+        vMACBProbePhy();
+
+        xMacInitStatus = eMACInitComplete;
+    }
+
+    if (eMACInitComplete != xMacInitStatus)
+    {
+        xResult = pdFAIL;
+    }
+    else
+    {
+        // Wait for the link status to come up before enabling interrupts
+        if (xPhyObject.ulLinkStatusMask != 0U)
+        {
+            // Enable the Ethernet RX and TX interrupt source.
+            MAP_EMACIntEnable(EMAC0_BASE, (EMAC_INT_RECEIVE | EMAC_INT_TRANSMIT |
+                          EMAC_INT_TX_STOPPED | EMAC_INT_RX_NO_BUFFER |
+                          EMAC_INT_RX_STOPPED | EMAC_INT_PHY));
+
+            // Enable EMAC interrupts
+            MAP_IntEnable(INT_EMAC0);
+        }
+        else
+        {
+            xResult = pdFAIL;
+        }
+    }
+
+    return xResult;
+}
+
+BaseType_t xNetworkInterfaceOutput(NetworkBufferDescriptor_t * const pxNetworkBuffer, BaseType_t xReleaseAfterSend)
+{
+    BaseType_t success = pdTRUE;
+    tEMACDMADescriptor *dma_descriptor;
+
+    // As this driver is strictly zero-copy, assert that the stack does not call this function with
+    // xReleaseAfterSend as false
+    configASSERT(0 != xReleaseAfterSend);
+
+    dma_descriptor = &_tx_descriptors[_tx_descriptor_list.write];
+
+    // If the DMA controller still owns the descriptor, all DMA descriptors are in use, bail out
+    if (0 == (dma_descriptor->ui32CtrlStatus & DES0_RX_CTRL_OWN))
+    {
```

Comment:
As `ui32CtrlStatus` looks like an unsigned value, the MISRA checker will like to see a comparison with `0U` in stead of just `0`.